lib/mapper-accessors.js

```diff
@@ -111,11 +111,31 @@ module.exports = Mapper => {
     HeatingCoolingState : {
       Current : {
         // Not quite the correct mapping: Homey supports `auto` mode, HomeKit doesn't. We'll default to `HEAT` when `auto` is set.
-        get : value => Characteristic.CurrentHeatingCoolingState[ value === 'off' ? 'OFF' : value === 'cool' ? 'COOL' : 'HEAT' ]
+        get: (value, { device }) => {
+          const currentTemperature = device.capabilitiesObj.measure_temperature?.value;
+          const targetTemperature = device.capabilitiesObj.target_temperature?.value;
+          const currentValue = value !== 'auto'
+            ? value.toUpperCase()
+            : currentTemperature < targetTemperature
+              ? 'HEAT'
+              : currentTemperature > targetTemperature
+                ? 'COOL'
+                : 'OFF';
+          if (currentValue in Characteristic.CurrentHeatingCoolingState) {
+            return Characteristic.CurrentHeatingCoolingState[currentValue];
+          }
+          return Characteristic.CurrentHeatingCoolingState.OFF;
```

Comment:
@robertklep, FYI I've added some security to HomeKitty's code given that now `thermostat_mode`'s values can be customized with Homey firmware 12.0.1+